### PR TITLE
Separating publisher consent data from vendor consent data

### PR DESCRIPTION
**Pull request by skoklowski**

### CODE REVIEW

File: src/components/popup/details/purposes/purposes.jsx [hunk 1]
```diff
@@ -49,12 +51,18 @@ export default class Purposes extends Component {
...
 			} else {
 				selectPurpose(selectedPurpose.id, isSelected);
 			}
 		}
 	};
 
+	createHandleSelectPurpose = (isPublisher) => {
+		return (data) => this.handleSelectPurpose(data, isPublisher);
+	};
+
```
Comment (mvciekzvbek) [inline src/components/popup/details/purposes/purposes.jsx:65]: Could be even shorter:

```suggestion
createHandleSelectPurpose = (isPublisher) => (data) => this.handleSelectPurpose(data, isPublisher);
```

File: src/lib/store.js [hunk 2]
```diff
@@ -369,6 +369,24 @@ export default class Store {
...
+		else {
+			selectedStandardPurposeIds.delete(purposeId);
+		}
+		this.storeUpdate();
+	};
+
+	selectAllStandardPurposes = (isSelected) => {
+		const {purposes = []} = this.vendorList || {};
+		const operation = isSelected ? 'add' : 'delete';
+		purposes.forEach(({id}) => this.publisherConsentData.selectedStandardPurposeIds[operation](id));
```
Comment (mvciekzvbek) [inline src/lib/store.js:386]: This is nice :D

**Review by mvciekzvbek — APPROVED** [review]

**Comment by osk-8** [comment]
The code looks good, but I'm wondering about one small think. Maybe we should change the naming from `standardPurposes` to `publisherPurposes`? What `standard` means in terms of purposes? 

Edit:
Nevermind, now I understand that it is in relation with `customPurposes`

File: src/components/popup/details/purposes/purposes.jsx (hunk 1 above)
Comment (skoklowski) [reply src/components/popup/details/purposes/purposes.jsx:65]: But I'm not sure if it would be more readable ;)

**Pull request merged**